amplify/backend/function/expressServer/src/controllers/auth.js

```diff
@@ -0,0 +1,142 @@
+const User = require('../models/user.model');
+
+const getHost = require('../utils/Host');
+const getEmailAPIKEY = require('../utils/EMAIL_API_KEY');
+
+var SibApiV3Sdk = require('sib-api-v3-sdk');
+var defaultClient = SibApiV3Sdk.ApiClient.instance;
+
+// Configure API key authorization: api-key
+var apiKey = defaultClient.authentications['api-key'];
+
+/**
+ *
+ * some controllers here
+ *
+ */
+
+// @route POST api/auth/recover
+// @desc Recover Password - Generates token and Sends password reset email
+// @access Public
+const recoverPassword = async (req, res, next) => {
+  const API_KEY = await getEmailAPIKEY();
+  apiKey.apiKey = API_KEY;
+
+  const host = await getHost();
+  try {
+    let user = await User.findOne({ email: req.body.email });
```

Comment:
You are accesing the ```req.body.email``` before you know if it exists. Check first if email is undefined then send back a bad data message before trying to run database query. Have a look at some of other routes for undefined check.